### PR TITLE
chore(deps): resolve aws sso profile issues by updating aws-sdk-go@v1.38

### PR DESCRIPTION
**Pull request by nitrocode**
## what
- [x] Use latest github.com/aws/aws-sdk-go@v1.38

## why
- Resolve `AWS_PROFILE` with `aws sso` issues

## references
- Closes https://github.com/segmentio/chamber/issues/376

## commands

Before update
```bash
✗ AWS_PROFILE=test chamber list-services
Error: Failed to get secret store: SharedConfigErr: only one credential type may be specified per profile: source profile, credential source, credential process, web identity token, or sso
```

```bash
✗ go get -u github.com/aws/aws-sdk-go@v1.38
```

~Replace was necessary in order to run `go get -u`~
```bash
✗ go mod edit -replace gopkg.in/segmentio/analytics-go.v3=github.com/segmentio/analytics-go/v3@v3.2.1
✗ go mod tidy
```

After update
```bash
✗ rm -f chamber && make build
✗ AWS_PROFILE=test ./chamber list-services
Service
hello-world/secret
```

### CODE REVIEW

**Comment by nitrocode** [comment]
Could you folks please review when you get a chance?

cc: @mckern @asaf-erlich @rikez

I think adding something like renovatebot would help out a lot with automating these upgrades in the future :)

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 	github.com/stretchr/testify v1.2.2
-	github.com/xtgo/uuid v0.0.0-20140804021211-a0b114877d4c // indirect
-	golang.org/x/sys v0.0.0-20220614162138-6c1b26c55098
-	gopkg.in/segmentio/analytics-go.v3 v3.0.1
+	golang.org/x/sys v0.7.0
+	gopkg.in/segmentio/analytics-go.v3 v3.2.1
 	gopkg.in/yaml.v3 v3.0.1
 )
 
-go 1.13
```
Comment (asaf-erlich) [inline go.mod:21]: Oh wow I definitely recall @rikez doing work to update golang version in a bunch of places but it's possible the go.mod file was left still on the old version. Still, I'm no longer on the team and will have them confirm if they're comfortable jumping to 1.20 or prefer to stay a version behind or not.

**Comment by asaf-erlich** [comment]
Is `renovatebot` preferred over github's dependabot which I'm more used to? Just found this: https://javascript.plainenglish.io/automate-dependency-updates-by-renovate-not-by-dependabot-6efddd549a3e . Happy to help the team turn dependabot on for now as I've done it for a bunch of our repos.

**Comment by nitrocode** [comment]
> Is renovatebot preferred over github's dependabot which I'm more used to?

dependabot is certainly nice but it's not as configurable as renovatebot so I prefer the latter.

To add renovatebot, it would be adding the bot to the repo and then merging the default PR that it generates.
To add dependabot, it's just enabling the bot on the repo (as you know).

Perhaps dependabot for now would be easier to keep dependencies up to date?

File: go.mod (hunk 1 above)
Comment (asaf-erlich) [reply go.mod:21]: I highly suspect it was just a mistake to leave the go.mod file with the older version. But I could be wrong.

Comment (alecjacobs5401) [reply go.mod:21]: @asaf-erlich @nitrocode there are ongoing discussions about the state of `go 1.13` in the `go.mod` file

https://github.com/segmentio/chamber/pull/371

@knksmith57 has some context and we've been discussing internally, but unfortunately haven't prioritized. 

TL;DR - by updating the go version in `go.mod`, its a potentially breaking change for any GitHub repos that are consuming `chamber` as a dependency (which is an option).

**Comment by nitrocode** [comment]
cc: @alecjacobs5401 please re-review when time permits

**Comment by alecjacobs5401** [comment]
@nitrocode I approved the workflow runs and it looks like things aren't happy for go1.16

**Comment by nitrocode** [comment]
Ok, I tested on go 1.16 and only updated the aws-sdk-go and that seems to have fixed the issue.

If I update to the latest aws-sdk-go, then it requires at least go 1.17 due to `x/sys@v0.1.0`

I was unable to test on 1.15 because it does not have an arm binary.

Please see if the new changes build across the different versions.

**Comment by nitrocode** [comment]
cc: @alecjacobs5401

**Review by alecjacobs5401 — APPROVED** [review]
LGTM - thanks for contributing this fix @nitrocode!

**Comment by nitrocode** [comment]
Thanks for merging @alecjacobs5401 ! What's the next step to get a new release out?

I'd imagine it would be 2.12.1 since it's a bug fix.

**Comment by nitrocode** [comment]
Oh I see this PR coming through https://github.com/segmentio/chamber/pull/382 so looks like soon 🙏

**Comment by alecjacobs5401** [comment]
@nitrocode I think we have a few things currently batched up so this might end up being a minor version bump. Should have a better answer/release for you by end of week

**Comment by nitrocode** [comment]
Thank you for releasing!!!
